ddtrace/contrib/pytest/_plugin_v2.py

```diff
@@ -402,10 +441,42 @@ def pytest_runtest_makereport(item, call) -> None:
         log.debug("encountered error during makereport", exc_info=True)
 
 
+@pytest.hookimpl(hookwrapper=True, tryfirst=True)
+def pytest_terminal_summary(terminalreporter, exitstatus, config):
+    """Report flaky or failed tests"""
+    # Before yield gives us a chance to show failure reports:
+    initial_size = len(terminalreporter.stats.get(PYTEST_STATUS.FAILED, []))
+
+    for failed_report in efd_get_failed_reports(terminalreporter):
+        failed_report.outcome = PYTEST_STATUS.FAILED
+        terminalreporter.stats.setdefault("failed", []).append(failed_report)
+
+    yield
+
+    # After yield gives us a chance to:
+    # - print our flaky test status summary
+    # - modify the total counts
+
+    # Reset the failed attempts so they don't count towards the final total
+    if initial_size == 0:
+        terminalreporter.stats.pop("failed", None)
+    else:
+        terminalreporter.stats[PYTEST_STATUS.FAILED] = terminalreporter.stats[PYTEST_STATUS.FAILED][:initial_size]
```

Comment:
I didn't get what the `initial_size` stuff is doing here.

Reply:
We append reports for failures from EFD retries to the original report of failures so that they can be displayed in the output, eg:
```
__________________ test_func_succeeds_after_5 __________________

    def test_func_succeeds_after_5():
        global _test_func_succeeds_after_5_count
        _test_func_succeeds_after_5_count += 1
>       assert _test_func_succeeds_after_5_count > 5
E       assert 5 > 5

tests/test_all_pass.py:97: AssertionError
```

However, if we leave them there, we'll change the count of failed items, and then overcount failures in https://github.com/DataDog/dd-trace-py/pull/11112/files#diff-a3f14c5d960a4c7bcb96a90bb3c7f853ec6e3e08afa999ee6814ebfb21a46afdR195-R202 .

I could just leave them there, and then not re-do the copy later (which is probably better for performance), but that feels even more disconnected.